File: 11-waterContainer/waterContainer.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        
        left = 0
        right = len(height) - 1
        maxArea = 0
        
        while left < right:
            lowerHeight =  height[left] if height[left] < height[right] else height[right]
            area = (right-left) * lowerHeight
            maxArea = maxArea if area < maxArea else area
            
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
            
        return maxArea

File: 1-100/11-20/11-waterContainer/waterContainer.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int volume = 0;
        
        while (left != right) {
            volume = max(volume, (right - left) * min(height[left], height[right]));
            if (height[left] <= height[right])
                ++left;
            else
                --right;
        }
        
        return volume;
    }
};